packages/backend/src/models/UserModel.ts

```diff
@@ -246,7 +284,11 @@ export class UserModel {
         if (user === undefined) {
             throw new NotFoundError(`Cannot find user with uuid ${userUuid}`);
         }
-        return mapDbUserDetailsToLightdashUser(user);
+
+        return mapDbUserDetailsToLightdashUser(
+            user,
+            await this.hasAuthentication(userUuid),
```

Comment:
A bigger refactor is needed to avoid calling the same static/private functions in most methods.